packages/astro/components/Debug.astro

```diff
@@ -1,306 +1,18 @@
-  const type = getType(node);
-  const props = {};
-
-  if (type === 'array' || type === 'object') {
-    props['data-char'] = type === 'array' ? ']' : '}'
-    props.open = !key && type === 'object' ? '' : undefined;
-  }
-  
-  return (
-    <details {...props} class={className}>
-        <Summary node={node} key={key} class={className} />
-        {children}
-    </details>
-  );
-}
-
-const Summary = ({ node, key, class: className }) => {
-  return (
-    <summary class={className}>{getSummary(node, key, className)}</summary>
-  );
-}
-
-const Empty = Symbol('Empty');
-
-const KeyValue = ({ key, value, dim, class: className }) => {
-  let type = key === '__proto__' ? 'prototype' : getType(value);
-  if (type === 'null') {
-    value = 'null';
-  } else if (type === 'undefined') {
-    value = 'undefined';
-  } else if (value === Empty) {
-    type = 'empty';
-    value = 'Empty';
-  } else {
-    value = JSON.stringify(value);
-  }
+import {format} from 'node:util';
+import {yellow, bold} from 'kleur/colors';
+import Code from './Code.astro';
 
-  return (
-    <div class={`${className} line`}>
-      {(key || !key && key === 0) && <><span class={`${className} key ${dim ? 'key-dim' : ''}`.trim()}>{key}</span><span class={`${className} sep`}>:</span></>}
-      <span class={`${className} value value-${type}`}>
-        {value}
-      </span>
-    </div>
-  )
+export interface Props {
+  [key: string]: any
 }
 
-const Node = ({ node, key, class: className, ...props }) => {
-  const type = getType(node);
-  className = className.replace(/debug-value/g, '');
-
-  if (type === 'array' || type === 'object') {
-    let children = [];
-    if (type === 'array' && node.length > 0 && Object.entries(node).length === 0) {
-      children = Array.from({ length: node.length }, (_, key) => <Node node={Empty} key={key} class={className} />);
-    } else {
-      children = Object.entries(node).map(([key, value]) => <Node node={value} key={key} class={className} />);
-    }
-    return (
-      <Details node={node} key={key} children={children} class={className} />
-    );
-  } else if (type === 'function') {
-    return (
-      <Details node={node} key={key} class={className} children={
-      <>
-        <KeyValue key="name" value={node.name} dim={true} class={className} />
-        <KeyValue key="__proto__" value="Function" dim={true} class={className} />
-      </>
-      }/>
-    );
-  } else if (type === 'promise') {
-    return (
-      <Details node={node} key={key} class={className} children={
-        <KeyValue key="__proto__" value="Promise" dim={true} />
-      } />
-    );
-  }
-
-  return <KeyValue key={key} value={node} class={className} />;
-}
+// The first prop is both the key & value
+const key = Object.keys(Astro.props)[0];
+const value = format(Astro.props[key]);
+// Log message to console, as well as the browser.
+console.debug(`[${yellow(bold('debug'))}] `, {[key]: value});
```

Comment:
Ack I didn't actually test the file, so I didn't notice `const value = format(Astro.props[key]);`.

```suggestion
console.debug(`[${yellow(bold('debug'))}] `, {[key]: Astro.props[key]});
```

We want to only have *one* round of `utils.format()` 😅

Reply:
It might also make sense to move the `util.format` call into the `<Code/>` template string, and just do `<Code lang="js" code={`// <Debug ${key}={...} />\n${format(value)}`} />`

Reply:
yup, lgtm!